Source/WPEFramework/Controller.cpp

```diff
@@ -822,13 +822,18 @@ namespace Plugin {
 
     Core::hresult Controller::Register(Exchange::Controller::ILifeTime::INotification* notification)
     {
+        ASSERT(notification != nullptr);
+
         _adminLock.Lock();
 
         // Make sure a sink is not registered multiple times.
-        ASSERT(std::find(_lifeTimeObservers.begin(), _lifeTimeObservers.end(), notification) == _lifeTimeObservers.end());
+        LifeTimeNotifiers::iterator index(std::find(_lifeTimeObservers.begin(), _lifeTimeObservers.end(), notification));
+        ASSERT(index == _lifeTimeObservers.end());
 
-        _lifeTimeObservers.push_back(notification);
-        notification->AddRef();
+        if (index == _lifeTimeObservers.end()) {
+            _lifeTimeObservers.push_back(notification);
+            notification->AddRef();
+        }
```

Comment:
In the return value, reflect that the Sync was not really added. In debug it will ASSERT, in Release it will at least report it..

Reply:
done